### ISSUE TITLE
gradient of LSTM in DeepAR is too tiny

### ISSUE THREAD

**Issue by innocezz** [jdb78/pytorch-forecasting#1294]
- PyTorch-Forecasting version: 1.0.0
- PyTorch version: 2.0.0
- Python version: 3.9
- Operating System: macos

### Expected behavior

I used the  data provided by get_stallion_data() and the deepar model to predict the volume in valid loader.  

### Actual behavior

i found that the training loss didn't show a downward trend and has been fluctuating. I use the function of log_gradient_flow to print the avg gradient of each layer,  found that the gradient of lstm is too tiny compared to other layers,  and the loss is fluctuating, all the code is provided by the official examples.

i hope someone could help me, why the loss didn't show a downward trend  and why the gradient of lstm is so small, i am not sure whether these are normal phenomenon? 

![gradient](https://user-images.githubusercontent.com/115786167/233586888-815f1abd-7389-4946-8c34-f95dea9e9bcd.jpg)
![loss](https://user-images.githubusercontent.com/115786167/233586922-3a04c62d-af16-42ec-ab4f-e2ae34afbfd1.jpg)

### Code to reproduce the problem

batch_size = 32  # set this between 32 to 128
train_dataloader = training.to_dataloader(train=True, batch_size=batch_size, num_workers=0)
val_dataloader = validation.to_dataloader(train=False, batch_size=batch_size, num_workers=0)

num_train_batch = len(train_dataloader)
num_eval_batch = len(val_dataloader)

# todo: loader中的target_scale能达到-3.7530e+15，是求loc和scale时出了问题：原因:clamp(log(0))趋向于负无穷
# todo: 只调lstm层的参数
# for x, y in train_dataloader:
#     print(x["target_scale"], "\n")

early_stop_callback = EarlyStopping(monitor="val_loss", min_delta=1e-4, patience=5, verbose=False, mode="min")

lr_logger = LearningRateMonitor()

trainer = pl.Trainer(
    max_epochs=10,
    accelerator="cpu",
    devices="auto",
    gradient_clip_val=1,
    # fast_dev_run=True,
    # logger=logger,
    # profiler=True,
    callbacks=[lr_logger, early_stop_callback],
)

deepar = DeepAR.from_dataset(
    training,
    learning_rate=1,
    hidden_size=32,
    dropout=0.1,
    # loss=NormalDistributionLoss(),
    loss=LogNormalDistributionLoss(),
    log_interval=200,
    log_val_interval=100,
    log_gradient_flow=True,
    # reduce_on_plateau_patience=3,
    num_train_batch=num_train_batch,
    num_eval_batch=num_eval_batch
)
print(f"Number of parameters in network: {deepar.size()/1e3:.1f}k")

# find optimal learning rate
deepar.hparams.log_interval = -1
deepar.hparams.log_val_interval = -1
trainer.limit_train_batches = 1.0
res = Tuner(trainer).lr_find(
    deepar, train_dataloaders=train_dataloader, val_dataloaders=val_dataloader, min_lr=1e-5, max_lr=1e2
)

print(f"suggested learning rate: {res.suggestion()}")
fig = res.plot(show=True, suggest=True)
fig.show()
deepar.hparams.learning_rate = res.suggestion()

torch.set_num_threads(10)
trainer.fit(
    deepar,
    train_dataloaders=train_dataloader,
    val_dataloaders=val_dataloader,
)

# calcualte mean absolute error on validation set
actuals = torch.cat([y for x, (y, weight) in iter(val_dataloader)])
predictions = deepar.predict(val_dataloader)
print(f"Mean absolute error of model: {(actuals - predictions).abs().mean()}")

# plot actual vs. predictions
prediction = deepar.predict(val_dataloader, mode="raw", return_x=True)
raw_predictions, x = prediction.output, prediction.x
for idx in range(10):  # plot 10 examples
    figure = deepar.plot_prediction(x, raw_predictions, idx=idx, add_loss_to_title=True)
plt.show()

Paste the command(s) you ran and the output. Including a link to a colab notebook will speed up issue resolution.
If there was a crash, please include the traceback here.
The code used to initialize the TimeSeriesDataSet and model should be also included.
